docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,223 @@
+specify the Task Queue that you're targeting, the Build ID that you're adding (or
+promoting), whether it becomes the new default version, and any existing versions it should be
+considered compatible with.
+
+The rest of this section uses updates to one Task Queue's version sets as examples.
+
+By default, both Task Queues and Workers are in an unversioned state. [Unversioned Worker](#unversioned-workers) can poll
+unversioned Task Queues and receive tasks. To use this feature, both the Task Queue and the Worker
+have to have Build IDs associated with them.
+
+If you run a Worker using versioning against a Task Queue that has not been set up to use versioning (or is missing that Worker's Build ID), it won't
+get any tasks. Likewise, a unversioned Worker polling a Task Queue with versioning won't work either.
+
+:::note Versions don't need to follow semver or any other semantic versioning scheme!
+
+The versions in the following examples look like semver versions for clarity, but they don't need to be.
+Versions can be any arbitrary string.
+
+:::
+
+First, add a version `1.0` to the Task Queue as the new default.
+Your version sets now look like this:
+
+| set 1 (default) |
+| --------------- |
+| 1.0 (default)   |
+
+All new Workflows started on the Task Queue have their first tasks assigned to version `1.0`.
+Workers with their Build ID set to `1.0` receive these Tasks.
+
+If Workflows that don't have an assigned version are still running on the Task Queue, Workers
+without a version, take those tasks. So ensure that such Workers are still operational if any
+Workflows were open when you added the first version. If you deployed any Workers with a _different_
+version, those Workers receive no Tasks.
+
+Now, imagine you need change the Workflow for some reason.
+
+Add `2.0` to the sets as the new default:
+
+| set 1         | set 2 (default) |
+| ------------- | --------------- |
+| 1.0 (default) | 2.0 (default)   |
+
+All new Workflows started on the Task Queue have their first tasks assigned to version `2.0`.
+Existing `1.0` Workflows keep generating tasks targeting `1.0`.
+Each deployment of Workers receives their respective Tasks.
+This same concept carries forward for each new incompatible version.
+
+Maybe you have a bug in `2.0`, and you want to make sure all open `2.0` Workflows switch to some new
+code as fast as possible. So, you'll add `2.1` to the sets, marking it as compatible with `2.0`. Now your
+sets look like this:
+
+| set 1         | set 2 (default) |
+| ------------- | --------------- |
+| 1.0 (default) | 2.0             |
+|               | 2.1 (default)   |
+
+All new Workflow Tasks that are generated for Workflows whose last Workflow Task completion was on
+version `2.0` are now assigned to version `2.1`. Because you specified that `2.1` is compatible with
+`2.0`, Temporal Server assumes that Workers with this version can process the existing Event
+Histories successfully.
```

Comment:
```suggestion
All new Workflow Tasks that are generated for Workflows whose last Workflow Task completion was on version `2.0` are now assigned to version `2.1`.
Because you specified that `2.1` is compatible with `2.0`, Temporal Server assumes that Workers with this version can process the existing Event Histories successfully.
```